packages/webapp/src/hooks/useOfflineCheck.ts

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from "react";
+import { toast } from "sonner";
+
+async function performNetworkCheck() {
+  const endpoint = "https://www.google.com";
+  const response = await fetch(endpoint, {
+    method: "HEAD",
+    cache: "no-cache",
+    mode: "no-cors",
+  });
+  return response.ok;
+}
+
+function useOfflineCheck(options: any = {}) {
+  const {
+
+    maxRetries = 3, // Maximum number of retries before assuming offline
+    retryDelay = 1500, // Delay between retries (ms)
+  } = options;
+
+  const [retries, setRetries] = useState(0);
+  
+
+  useEffect(() => {
+    const handleOffline = () => {
+      if (retries < maxRetries) {
+        setTimeout(() => {
+          performNetworkCheck()
+            .then(() => {
+              setRetries(0);
+              toast.dismiss();
+            }) // online
+            .catch(() => {
+              const newRetry = retries + 1;
+              setRetries(newRetry); // Retry if error occurs
+            });
+        }, retryDelay);
+      } 
+    };
+
+    window.addEventListener("online", handleOffline);
```

Comment:
It seems unecessary to recheck the network state when receiving "online".

Reply:
this is checked to dismiss the toast.